docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,81 @@
+
+#### Habitat Client
+The `habitatctl` CLI can be used to control Habitat remotely or locally. It provides commands that interface directly with the Habitat API. It gives the option for custom addresses, as well as the ability to hit the API via `libp2p` to circumvent NATs.
+
+In the future, a more interactive CLI can be created using the Bubble Tea library. In addition, `habitatctl` can run a local server for a client web app to make interacting with Habitat easier for non-technical users.
+
+#### Community Management
+A community is a cluster of nodes that collectively maintain a shared filesystem, shared applications, and more. These nodes can be spread out geographically and can number from one to thousands. Communities have a consistent state that is maintained by a subset of the community nodes executing the [Raft](https://raft.github.io/) consensus algorithm. The community control layer is executed by the `CommunityManager` struct.
+
+##### Consensus
+All nodes need to agree on the community state for Habitat to function. To reach agreement on the state, nodes in a community run the Raft algorithm to create an ordered log of state updates. This effectively turns the community into a replicated state machine. Since Raft does not scale well beyond ~10ish nodes (although I don't know if this has been tested at large scales), larger communities will only have some nodes elected as Raft participants. When a new state is agreed upon, the new updates will be broadcast to all other nodes. A algorithm needs to be created to determine what nodes are Raft nodes.
+
+
+
+##### State Management
+Community state is represented as a JSON object conforming to a JSON schema. There is a limited set of allowed state transitions defined in `transitions.go` in the `state` package. Each transition is defined by a JSON patch that specifies the changes to the community state, as well as a validator function that checks the transition and the resulting community state for errors. Instances of the `Executor` interface receive these state updates via a channel, and can act on them. Each node service should implement this interface to be able to react to these updates.
+
+When a node restarts, all of the layer two node services are reset to the last known state for each community. Then, it asks the community for state updates it missed, and each is applied individually. After that, the node will receive state updates normally.
+
+### Applications
+Applications are the top level of Habitat. Applications can contain server and web content portions. They are started by the process manager based off of community state.
+
+In the future, a SDK will be provided to make developing Habitat applications easier. In addition, a package manager will be provided that can be used to install and update Habitat apps.
```

Comment:
nit: will Habitat own the package manager? or will people be able to create their own? (i forget what we decided from brainstorming)